File: src/Algorithms/Closest_Value_in_BST/closest-value-in-bst.test.ts
// // @ts-nocheck

// import { TestSolutions } from "@utils/test-solutions";

// import { BinaryTree } from "../../Data_Structures/Binary_Tree/binary-tree";

// import { solutions } from "./closest-value-in-bst";

// describe("Find closest value in BST", () => {
//   const binaryTreeA = new BinaryTree(10);
//   const binaryTreeB = new BinaryTree(5);

//   for (const value of [5, 15, 2, 6, 13, 22, 1, 14]) {
//     binaryTreeA.insert(value);
//   }

//   for (const value of [2, 8, -2, 10, 12, 3, 15, -15, 21]) {
//     binaryTreeB.insert(value);
//   }

//   TestSolutions(solutions, (solution) => {
//     expect(solution(binaryTreeA, 12)).toBe(13);
//     expect(solution(binaryTreeA, 15)).toBe(15);
//     expect(solution(binaryTreeB, 7)).toBe(8);
//   });
// });

///================

///================

import { distance_path } from "./closest-value-in-bst";

test("", () => {
  expect(
    distance_path(0, 9, [123, 864, 783, 81, 6, 87, 94, 46, 1, 14])
  ).toEqual({
    minCost: 23,
    minPath: "01469",
  });
});

File: src/Algorithms/Trees/Closest_Value_in_BST/closest_value_in_bst_test.py
from closest_value_in_bst import findClosestValueValueRecursively, findClosestValueValueIteratively

from src.Data_Structures.Binary_Tree.binary_tree import BinaryTree

binaryTreeA = BinaryTree(10)
binaryTreeB = BinaryTree(5)

list = [5, 15, 2, 6, 13, 22, 1, 14]
for value in list:
    binaryTreeA.insert(value)

list = [2, 8, -2, 10, 12, 3, 15, -15, 21]
for value in list:
    binaryTreeB.insert(value)


def runAssertions(solution):
    assert solution(binaryTreeA, 12) == 13
    assert solution(binaryTreeA, 15) == 15
    assert solution(binaryTreeB, 7) == 8


def test_findClosestValueValueRecursively():
    runAssertions(findClosestValueValueRecursively)


def test_findClosestValueValueIteratively():
    runAssertions(findClosestValueValueIteratively)
